src/ptrace_utils.c

```diff
@@ -133,8 +133,12 @@ void read_data(int pid, void *addr, void *vptr, int len)
 }
 
 void read_lwps(char *arg_pid) {
-	int pid = atoi(arg_pid);
-	
+        char *pEnd;
+        long int pid = strtol(arg_pid, &pEnd, 10);
```

Comment:
The same comment as scan_mem except that apparently nothing calls this?

Reply:
It was implemented when I started looking at kernel threads and hence trying to see what ptrace could give us. I haven't worked further on this yet, will extend this piece of code when I get back to it.

Reply:
Addressed and move the error handling to the caller (when it's implemented).